### PR TITLE
Retire the Core_A folder and use a single Core folder for all processor core headers

### PR DESCRIPTION
**Pull request by Masmiseim36**
* Separate folders for different architecture profiles (A-profile, R-profile and M-profile)
* Top-level compiler headers will exist in the Core folder (no separate Compiler folder is needed). ** The cmsis_compiler.h header will continue to figure out which compiler toolchain is being used. ** Compiler headers specific to each architecture profiles can reside within the new architecture profile folders. ** The second-level files like cmsis_gcc.h are named cmsis_gcc_a.h, cmsis_gcc_r.h and cmsis_gcc_m.h to avoid having multiple files with the same name

### CODE REVIEW

**Review by JonatanAntoni — CHANGES_REQUESTED** [review]
The `core_armv8xxx.h` are effectively core header files (even if not implementing a specific core but only architecture generic features) and need to stay on top level. As they do not relate to any specific core, we might decide to remove them from CMSIS 6 (@GuentherMartin, what do you think?).

**Comment by GuentherMartin** [comment]
In my opinion we should delete the core_armv8xxx.h core header files.
These are the only header files representing an architecture and not a dedicated core. The files were added because no specific ARM v8-M device was available when Arm v-M support was added.

File: CMSIS/Core/Include/a-profile/cmsis_armcc_a.h
Comment (GuentherMartin) [inline CMSIS/Core/Include/a-profile/cmsis_armcc_a.h:1]: In my opinion this file is obsolete.
Arm Compiler V5 (ARMCC) is no longer used

File: CMSIS/Core/Include/m-profile/core_armv81mml.h
Comment (GuentherMartin) [inline CMSIS/Core/Include/m-profile/core_armv81mml.h:1]: In my opinion this file should be deleted.

File: CMSIS/Core/Include/m-profile/core_armv8mbl.h
Comment (GuentherMartin) [inline CMSIS/Core/Include/m-profile/core_armv8mbl.h:1]: In my opinion this file should be deleted.

File: CMSIS/Core/Include/m-profile/core_armv8mml.h
Comment (GuentherMartin) [inline CMSIS/Core/Include/m-profile/core_armv8mml.h:1]: In my opinion this file should be deleted.

**Review by GuentherMartin — DISMISSED** [review]

**Comment by Masmiseim36** [comment]
Hello @JonatanAntoni and @GuentherMartin 

thank you for the feedback. I have updated the Pull-Request according to your input.

**Comment by Masmiseim36** [comment]
@JonatanAntoni  Is there anything I can do regarding the validation errors?

**Comment by JonatanAntoni** [comment]
> @JonatanAntoni Is there anything I can do regarding the validation errors?

No, unfortunately, to run the tests we need the license key which I stored in a repository secret. For security reasons a PR from a fork doesn't get access to these. Hence, we cannot execute the tests.

**Pull request merged**